pulsar-spark/src/main/java/com/yahoo/pulsar/spark/SparkStreamingPulsarReceiver.java

```diff
@@ -0,0 +1,100 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yahoo.pulsar.spark;
+
+import org.apache.spark.storage.StorageLevel;
+import org.apache.spark.streaming.receiver.Receiver;
+
+import com.yahoo.pulsar.client.api.ClientConfiguration;
+import com.yahoo.pulsar.client.api.Consumer;
+import com.yahoo.pulsar.client.api.ConsumerConfiguration;
+import com.yahoo.pulsar.client.api.Message;
+import com.yahoo.pulsar.client.api.MessageListener;
+import com.yahoo.pulsar.client.api.PulsarClient;
+import com.yahoo.pulsar.client.api.PulsarClientException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SparkStreamingPulsarReceiver extends Receiver<String> {
+
+    private ConsumerConfiguration consumerConfiguration;
+    private PulsarClient pulsarClient;
+    private String topic;
+    private String subscription;
+
+    public SparkStreamingPulsarReceiver(
+            ClientConfiguration clientConfiguration,
+            ConsumerConfiguration consumerConfiguration,
+            String url,
+            String topic,
+            String subscription
+    ) {
+        this(
+                StorageLevel.MEMORY_AND_DISK_2(),
+                clientConfiguration,
+                consumerConfiguration,
+                url,
+                topic,
+                subscription
+        );
+    }
+
+    public SparkStreamingPulsarReceiver(
+            StorageLevel storageLevel,
+            ClientConfiguration clientConfiguration,
+            ConsumerConfiguration consumerConfiguration,
+            String url,
+            String topic,
+            String subscription
+    ) {
+        super(storageLevel);
+        this.topic = topic;
+        this.subscription = subscription;
+        consumerConfiguration.setMessageListener((consumer, msg) -> {
+            store(new String(msg.getData()));
+            consumer.acknowledgeAsync(msg);
+        });
+        this.consumerConfiguration = consumerConfiguration;
+        try {
+            pulsarClient = PulsarClient.create(url, clientConfiguration);
+        } catch (PulsarClientException e) {
+            log.error("Failed to create client : {}", e.getMessage());
+        }
+    }
+
+    public void onStart() {
+        try {
+            pulsarClient.subscribe(topic, subscription, consumerConfiguration);
+        } catch (PulsarClientException e) {
+            log.error("Failed to start subscription : {}", e.getMessage());
+            restart("Restart a consumer");
+        }
+    }
+
+    public void onStop() {
+        try {
+            if (pulsarClient != null) {
```

Comment:
Sorry, I confirmed that after restart is called,  onStop() -> onStart() is called.
As we create a client in the constructor, it cannot start subscribing when restarting.
I think we should create a client in onStart().